packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -167,13 +189,49 @@ export async function isHotswappableEcsServiceChange(
   return ret;
 }
 
+function mergeTaskDefinitions(patch: {containerDefinitions: {[key:string]: any}[]}, target: AWS.ECS.DescribeTaskDefinitionResponse) {
+  const src = patch.containerDefinitions;
+  const dst = target.taskDefinition?.containerDefinitions;
+  if (dst === undefined) {
+    return;
+  }
+  // schema: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDefinition.html
+  for (const i in src) {
+    for (const key of Object.keys(src[i])) {
+      (dst[i] as any)[key] = src[i][key];
+    }
+  }
```

Comment:
Doesn't this need to be recursive?

Reply:
No, because we don't also recurse here: https://github.com/aws/aws-cdk/pull/26404#discussion_r1283073969